crates/libs/registry/src/key.rs

```diff
@@ -278,6 +287,40 @@ impl Key {
         }
     }
 
+    /// Gets the value for the name in the registry key.
+    pub fn get_hstring<T: AsRef<str>>(&self, name: T) -> Result<HSTRING> {
+        let name = pcwstr(name);
+        let mut ty = 0;
+        let mut len = 0;
+
+        let result = unsafe {
+            RegQueryValueExW(self.0, name.as_ptr(), null(), &mut ty, null_mut(), &mut len)
+        };
+
+        win32_error(result)?;
+
+        if !matches!(ty, REG_SZ | REG_EXPAND_SZ) {
```

Comment:
No change needed now but just wanted to be a nerd and point out that the registry has a much larger set of supported values than this crate currently supports. Would be great to relax this check one day and let developers, for example, set a value of type 0x1234 with hstring data.

Reply:
Why not call `get_bytes` instead of `get_hstring` since it doesn't have this check?

Reply:
Unless you're saying this check adds no value?

Reply:
The non standard registry typed value could still hold a string. So this check would prevent me from retrieving that string if the type is 0x1234. (Not a big deal now, we can add support for alien types in a new PR.)